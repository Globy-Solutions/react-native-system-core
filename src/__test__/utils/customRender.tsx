import {render} from '@testing-library/react-native';
import {ErrorBoundary} from 'react-error-boundary';
import renderer from 'react-test-renderer';
import ThemeProvider from 'src/theme/theme-provider';

type Props = {
  children: React.ReactNode;
};

const logError = (error: Error, info: {componentStack: string}) => {
  // Do something with the error, e.g. log to an external API
  console.log(error, info);
};
const Provider = ({children}: Props) => (
  // @ts-ignore
  <ErrorBoundary FallbackComponent="Something went wrong" onError={logError}>
    <ThemeProvider>{children}</ThemeProvider>
  </ErrorBoundary>
);
const App = ({children}: Props) => (
  // @ts-ignore
  <ErrorBoundary FallbackComponent="Something went wrong" onError={logError}>
    {children}
  </ErrorBoundary>
);
const toJSON = (component: any) => {
  const r = App(component);
  renderer.create(r).toJSON();
  return render(r);
};

export * from '@testing-library/react-native';
export {App, Provider, toJSON};
