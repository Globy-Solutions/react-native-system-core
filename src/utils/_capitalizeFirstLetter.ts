export const capitalize = (text: string) => {
  const firstLetter = text.charAt(0);
  const firstLetterCap = firstLetter.toUpperCase();
  const remainingLetters = text.slice(1);
  return firstLetterCap + remainingLetters;
};

const capitalizeFirstLetter = (text: string) => {
  const textArr = /\b/i.test(text);
  if (textArr) {
    const arr = text.split(' ');
    return [arr[0], arr[1]]
      .map(t => capitalize(t as string).charAt(0))
      .join('');
  } else {
    return capitalize(text);
  }
};

export default capitalizeFirstLetter;
