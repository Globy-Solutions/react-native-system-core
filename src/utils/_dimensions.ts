import {Dimensions, PixelRatio} from 'react-native';

import type {DimensionValue} from 'react-native/Libraries/StyleSheet/StyleSheetTypes';

const {height, width}: {height?: DimensionValue; width?: DimensionValue} =
  Dimensions.get('screen');

const LINE_HEIGHT_SCALE_FACTOR = 1.33;

/**
 * Converts provided width percentage to independent pixel (dp).
 * @param  {string} widthPercent The percentage of screen's width that UI element should cover
 *                               along with the percentage symbol (%).
 * @return {number}              The calculated dp depending on current device's screen width.
 */
const widthPercentageToDP = (widthPercent: string | number) => {
  // Parse string percentage input and convert it to number.
  const elemWidth =
    typeof widthPercent === 'number' ? widthPercent : parseFloat(widthPercent);
  // Use PixelRatio.roundToNearestPixel method in order to round the layout
  // size (dp) to the nearest one that correspons to an integer number of pixels.
  return PixelRatio.roundToNearestPixel((width * elemWidth) / 100);
};

/**
 * Converts provided height percentage to independent pixel (dp).
 * @param  {string} heightPercent The percentage of screen's height that UI element should cover
 *                                along with the percentage symbol (%).
 * @return {number}               The calculated dp depending on current device's screen height.
 */
const heightPercentageToDP = (heightPercent: string | number): number => {
  // Parse string percentage input and convert it to number.
  const elemHeight =
    typeof heightPercent === 'number'
      ? heightPercent
      : parseFloat(heightPercent);

  // Use PixelRatio.roundToNearestPixel method in order to round the layout
  // size (dp) to the nearest one that correspons to an integer number of pixels.
  return PixelRatio.roundToNearestPixel((height * elemHeight) / 100);
};

/*
* Implement that piece of code to have orientation detection

Dimensions.get('screen').width < Dimensions.get('screen').height ? 'portrait' : 'landscape'
*/

const guidelineBaseWidth: number = 375;
const guidelineBaseHeight: number = 667;
const horizontalScale = (size: number) => (width / guidelineBaseWidth) * size;
const verticalScale = (size: number) => (height / guidelineBaseHeight) * size;
const moderateScale = (size: number, factor = 0.5) =>
  size + (horizontalScale(size) - size) * factor;

const getFontLineHeight = (fontSize: number) =>
  fontSize * LINE_HEIGHT_SCALE_FACTOR;

export {
  getFontLineHeight,
  height,
  horizontalScale,
  heightPercentageToDP as hp,
  moderateScale,
  verticalScale,
  width,
  widthPercentageToDP as wp
};
