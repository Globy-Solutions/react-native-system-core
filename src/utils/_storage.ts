/**
 * Method for LocalStorage
 * setData allways return Array object
 *
 * @function
 */

import {MMKV} from 'react-native-mmkv';

const AsyncStorage = new MMKV();

const getData = (name: string) => {
  const value = AsyncStorage.getString(name);
  return value ? JSON.parse(value) : null;
};

const setData = (name: string, data: any): void => {
  const currentState = getData(name);
  let temp: any[] = [];
  if (currentState) {
    temp = [...currentState, data];
  } else {
    temp = [data];
  }
  AsyncStorage.set(name, JSON.stringify(temp));
};

const deleteData = (name: string) => {
  AsyncStorage.delete(name);
};

export {deleteData, getData, setData};
