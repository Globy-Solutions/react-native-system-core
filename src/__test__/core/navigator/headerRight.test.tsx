import {Text} from 'react-native';

import HeaderRight from 'src/navigator/header/right';

import {Provider, render, screen} from 'src/__test__/utils/customRender';
// import styles from 'src/navigator/header/header-styles';
// import { wp } from 'src/utils/_dimensions';

import type {ReactTestInstance} from 'react-test-renderer';

const Header = () => (
  <Provider>
    <HeaderRight>
      <Text>Test Child</Text>
    </HeaderRight>
  </Provider>
);

const renderHeader = () => {
  render(<Header />);
  return screen.getByText('Test Child');
};
/* 
const updateStyleBasedOnOrientation = (orientation: string) => {
  renderHeader();
  const headerRight = screen.getByText('Test Child');
  const container = headerRight.parent?.parent;
  jest.mock('src/utils/useOrientation', () => ({
    useOrientation: () => ({ orientation })
  }));

  return container?.props.style;
};
 */
describe('HeaderRight', () => {
  describe('Render', () => {
    let headerRight: ReactTestInstance;
    beforeEach(() => {
      headerRight = renderHeader();
    });
    test('snapshot', () => {
      expect(screen).toMatchSnapshot();
    });
    test('renders children correctly', () => {
      expect(headerRight).toBeTruthy();
    });
  });
  /* 
  describe('Styles', () => {
    test('updates style based on orientation portrait', () => {
      const style = updateStyleBasedOnOrientation('portrait');
      expect(style).toEqual([
        styles.tapbarRight,
        { width: wp(50) }
      ]);
    });
    test('updates style based on orientation landscape', () => {
      const style = updateStyleBasedOnOrientation('landscape');
      expect(style).toEqual([
        styles.tapbarRight,
        { width: wp(70) }
      ]);
    });
  });
   */
});
