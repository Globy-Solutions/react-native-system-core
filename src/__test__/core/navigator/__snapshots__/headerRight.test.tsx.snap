// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderRight Render snapshot 1`] = `
<View
  style={
    [
      {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "maxHeight": 40,
        "minHeight": 40,
        "paddingRight": 10,
      },
      {
        "width": 375,
      },
    ]
  }
>
  <Text>
    Test Child
  </Text>
</View>
`;
